packages/atomic/scripts/check-light-dom-host-selectors.mjs

```diff
@@ -0,0 +1,322 @@
+#!/usr/bin/env node
+
+/**
+ * Static analysis script to detect invalid `:host` selectors in Light DOM components.
```

Comment:
I haven't the code of the check in detail, but the general idea LGTM. 
It doesn't match any cases at the moment, which is good.
Have we tested that it matches violations properly?